oracle/oracle.go

```diff
@@ -169,20 +171,22 @@ func WorkOracle(opts ...OptionFunc) (WorkOracleResult, error) {
 }
 
 // CreateBlockCipher creates an AES cipher for given fast oracle block.
-// A cipher is created using an idx encrypted with challenge:
-//
-//	cipher = AES(AES(ch).Encrypt(i))
-func CreateBlockCipher(ch shared.Challenge, nonce uint8) (cipher.Block, error) {
-	// A temporary cipher used only to create key.
-	// The key is a block encrypted with AES which key is the challenge.
-	keyCipher, err := aes.NewCipher(ch)
-	if err != nil {
-		return nil, err
+// A cipher is created using an idx encrypted with challenge.
+func CreateBlockCipher(ch shared.Challenge, nonceGroup uint32, k2Pow uint64) (cipher.Block, error) {
+	hasher := blake3.New(16, nil)
+	hasher.Write(ch)
+
+	nonceBytes := make([]byte, 4)
+	binary.LittleEndian.PutUint32(nonceBytes, nonceGroup)
+	hasher.Write(nonceBytes)
+
+	k2PowBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(k2PowBytes, k2Pow)
+	hasher.Write(k2PowBytes)
```

Comment:
This is shorter and saves on allocating temporary memory:

```suggestion
	binary.Write(hasher, binary.LittleEndian, ch)
	binary.Write(hasher, binary.LittleEndian, nonceGroup)
	binary.Write(hasher, binary.LittleEndian, k2Pow)
```